xbmc/video/windows/GUIWindowVideoNav.cpp

```diff
@@ -268,6 +268,112 @@ std::string CGUIWindowVideoNav::GetQuickpathName(const std::string& strPath) con
   }
 }
 
+int CGUIWindowVideoNav::SelectFirstUnwatchedItem()
+{
+  if (m_vecItems->IsVideoDb())
+  {
+    NODE_TYPE NodeType = CVideoDatabaseDirectory::GetDirectoryChildType(m_vecItems->GetPath());
+
+    if (NodeType == NODE_TYPE_SEASONS || NodeType == NODE_TYPE_EPISODES)
+      return CSettings::Get().GetInt("videolibrary.tvshowselectfirstunwatcheditem");
+  }
+
+  return 0;
+}
+
+int CGUIWindowVideoNav::IncludeAllSeasonsAndSpecials()
+{
+  return CSettings::Get().GetInt("videolibrary.tvshowsincludeallseasonsandspecials");
+}
+
+int CGUIWindowVideoNav::GetFirstUnwatchedItemIndex(bool includeAllSeasons, bool includeSpecials)
+{
+  int iIndex = 0;
+
+  int iUnwatchedSeason = INT_MAX;
+
+  // Run through the list of items and find the season number of the first season with unwatched episodes
+  for (int i = 0; i < m_vecItems->Size(); ++i)
```

Comment:
`m_vecItems` use a zero-based index, we shouldn't increment it before, so this should be `i++`